hypothesis-python/src/hypothesis/internal/conjecture/data.py

```diff
@@ -994,52 +994,64 @@ def draw_integer(
         shrink_towards: int = 0,
         forced: Optional[int] = None,
     ) -> int:
+        if min_value is not None:
+            shrink_towards = max(min_value, shrink_towards)
+        if max_value is not None:
+            shrink_towards = min(max_value, shrink_towards)
+
```

Comment:
I *think* that clamping `shrink_towards` at the start instead of special casing `shrink_towards <= min_value` and `max_value <= shrink_towards` is equivalent, and much simpler to implement forcing for.

However, I'm not sure if there are performance implications here. Before, we didn't probe (potentially multiple times thanks to rejection sampling) in the cases where `shrink_towards` was out of bounds. Now we do. I'm guessing this is pretty much fine though, especially since out of bounds `shrink_towards` seem pretty rare. 

Even if we disallow out of bounds `shrink_towards` disallow via `assert`, call sites would effectively move to the behavior in this pr regardless, since clamping to `{min, max}_value` is the only reasonable thing to do in that scenario. I think implicitly clamping for call sites is probably fine to avoid boilerplate, though it may catch a few accidental typos making `shrink_towards` too large or small.

Reply:
Seems fine; my only concern is whether this is linked to the `nocover` test which is failing about the distribution of integers in very wide ranges - we'll want to keep the status quo for that.